File: bitget-node-sdk-api/src/lib/v1/SpotOrderApi.ts
import {BaseApi} from '../BaseApi';

export class SpotOrderApi extends BaseApi {

    orders(qsOrBody: object) {
        const url = '/api/spot/v1/trade/orders';
        const headers = this.signer('POST', url, qsOrBody)
        return this.axiosInstance.post(url, qsOrBody, {headers})
    }

    batchOrders(qsOrBody: object) {
        const url = '/api/spot/v1/trade/batch-orders';
        const headers = this.signer('POST', url, qsOrBody)
        return this.axiosInstance.post(url, qsOrBody, {headers})
    }

    cancelOrder(qsOrBody: object) {
        const url = '/api/spot/v1/trade/cancel-order';
        const headers = this.signer('POST', url, qsOrBody)
        return this.axiosInstance.post(url, qsOrBody, {headers})
    }

    cancelBatchOrder(qsOrBody: object) {
        const url = '/api/spot/v1/trade/cancel-batch-orders';
        const headers = this.signer('POST', url, qsOrBody)
        return this.axiosInstance.post(url, qsOrBody, {headers})
    }

    openOrders(qsOrBody: object) {
        const url = '/api/spot/v1/trade/open-orders';
        const headers = this.signer('POST', url, qsOrBody)
        return this.axiosInstance.post(url, qsOrBody, {headers})
    }

    history(qsOrBody: object) {
        const url = '/api/spot/v1/trade/history';
        const headers = this.signer('POST', url, qsOrBody)
        return this.axiosInstance.post(url, qsOrBody, {headers})
    }

    fills(qsOrBody: object) {
        const url = '/api/spot/v1/trade/fills';
        const headers = this.signer('POST', url, qsOrBody)
        return this.axiosInstance.post(url, qsOrBody, {headers})
    }

    placePlanOrder(qsOrBody: object) {
        const url = '/api/spot/v1/plan/placePlan';
        const headers = this.signer('POST', url, qsOrBody)
        return this.axiosInstance.post(url, qsOrBody, {headers})
    }

    cancelPlanOrder(qsOrBody: object) {
        const url = '/api/spot/v1/plan/cancelPlan';
        const headers = this.signer('POST', url, qsOrBody)
        return this.axiosInstance.post(url, qsOrBody, {headers})
    }

    currentPlanOrder(qsOrBody: object) {
        const url = '/api/spot/v1/plan/currentPlan';
        const headers = this.signer('POST', url, qsOrBody)
        return this.axiosInstance.post(url, qsOrBody, {headers})
    }

    historyPlanOrder(qsOrBody: object) {
        const url = '/api/spot/v1/plan/historyPlan';
        const headers = this.signer('POST', url, qsOrBody)
        return this.axiosInstance.post(url, qsOrBody, {headers})
    }

    traderOrderCloseTracking(qsOrBody: object) {
        const url = '/api/spot/v1/trace/order/closeTrackingOrder';
        const headers = this.signer('POST', url, qsOrBody)
        return this.axiosInstance.post(url, qsOrBody, {headers})
    }

    traderOrderCurrentTrack(qsOrBody: object) {
        const url = '/api/spot/v1/trace/order/orderCurrentList';
        const headers = this.signer('POST', url, qsOrBody)
        return this.axiosInstance.post(url, qsOrBody, {headers})
    }

    traderOrderHistoryTrack(qsOrBody: object) {
        const url = '/api/spot/v1/trace/order/orderHistoryList';
        const headers = this.signer('POST', url, qsOrBody)
        return this.axiosInstance.post(url, qsOrBody, {headers})
    }
}

File: bitget-java-sdk-api/src/main/java/com/bitget/openapi/api/v1/SpotOrderApi.java
package com.bitget.openapi.api.v1;

import com.bitget.openapi.dto.response.ResponseResult;
import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.POST;

import java.util.Map;

public interface SpotOrderApi {

    // order
    @POST("/api/spot/v1/trade/orders")
    Call<ResponseResult> orders(@Body Map<String, String> body);

    @POST("/api/spot/v1/trade/batch-orders")
    Call<ResponseResult> batchOrders(@Body Map<String, String> body);

    @POST("/api/spot/v1/trade/cancel-order")
    Call<ResponseResult> cancelOrder(@Body Map<String, String> body);

    @POST("/api/spot/v1/trade/cancel-batch-orders")
    Call<ResponseResult> cancelBatchOrder(@Body Map<String, String> body);

    @POST("/api/spot/v1/trade/open-orders")
    Call<ResponseResult> openOrders(@Body Map<String, String> body);

    @POST("/api/spot/v1/trade/history")
    Call<ResponseResult> history(@Body Map<String, String> body);

    @POST("/api/spot/v1/trade/fills")
    Call<ResponseResult> fills(@Body Map<String, String> body);


    // plan
    @POST("/api/spot/v1/plan/placePlan")
    Call<ResponseResult> placePlan(@Body Map<String, String> body);

    @POST("/api/spot/v1/plan/cancelPlan")
    Call<ResponseResult> cancelPlan(@Body Map<String, String> body);

    @POST("/api/spot/v1/plan/currentPlan")
    Call<ResponseResult> currentPlan(@Body Map<String, String> body);

    @POST("/api/spot/v1/plan/historyPlan")
    Call<ResponseResult> historyPlan(@Body Map<String, String> body);


    // trace
    @POST("/api/spot/v1/trace/order/closeTrackingOrder")
    Call<ResponseResult> traderCloseTrackingOrder(@Body Map<String, String> body);

    @POST("/api/spot/v1/trace/order/orderCurrentList")
    Call<ResponseResult> traderOrderCurrentList(@Body Map<String, String> body);

    @POST("/api/spot/v1/trace/order/orderHistoryList")
    Call<ResponseResult> traderOrderHistoryList(@Body Map<String, String> body);
}
